### PR TITLE
add-jieba-pipeline

### PR DESCRIPTION
**Pull request by LegendreM**

### CODE REVIEW

File: src/tokenizer.rs [hunk 1]
```diff
@@ -125,25 +139,44 @@ impl Analyzer {
...
         }
     }
+
+    /// Try to Detect Language and return the corresponding pipeline,
+    /// if no language is detected or no pipeline corresponds to the Language,
+    /// the function fallback in function `pipeline_from_script`
+    fn pipeline_from_lang<'a>(&'a self, text: &'a str) -> &Pipeline {
+        match detect_lang(text) {
+            Language::Other => self.pipeline_from_script(text),
+            language => &self.config.pipeline_lang_map.get(&language).unwrap_or(self.pipeline_from_script(text))
```
Comment (LegendreM) [inline src/tokenizer.rs:159]:
```suggestion
            language => &self.config.pipeline_lang_map.get(&language).unwrap_or_else(|| self.pipeline_from_script(text))
```

File: src/tokenizer.rs [hunk 2]
```diff
@@ -125,55 +134,107 @@ impl Analyzer {
...
+        let language = self.detect_lang(text);
+        self.config.pipeline_map.get(&(script, language))
+            .or_else(|| self.config.pipeline_map.get(&(script, Language::Other)))
+            .or_else(|| self.config.pipeline_map.get(&(Script::Other, Language::Other)))
+            .unwrap_or_else(|| &*DEFAULT_PIPELINE)
+    }
+
+    /// detect script with whatlang,
+    /// if no script is detected, return Script::Other
+    fn detect_script<'a>(&'a self, text: &'a str) -> Script {
```
Comment (MarinPostma) [inline src/tokenizer.rs:165]: I think lifetimes can be ellided here: the return value is not tied to self

File: src/tokenizer.rs [hunk 3]
```diff
@@ -125,55 +134,107 @@ impl Analyzer {
...
+        whatlang::detect_script(text).map(Script::from).unwrap_or(Script::Other)
+    }
+    
+    /// detect lang (dummy)
+    fn detect_lang<'a>(&'a self, text: &'a str) -> Language {
```
Comment (MarinPostma) [inline src/tokenizer.rs:170]: same here

File: src/tokenizer.rs [hunk 4]
```diff
@@ -125,55 +134,107 @@ impl Analyzer {
...
-        .map(Script::from)
-        .unwrap_or(Script::Other);
-    (script, Language::Other)
+    /// Try to Detect Language and Script and return the corresponding pipeline,
+    /// if no Language is detected or no pipeline corresponds to the Language
+    /// the function try to get a pipeline corresponding to the script;
+    /// if no Script is detected or no pipeline corresponds to the Script,
+    /// the function try to get the default pipeline in the map;
+    /// if no default pipeline exist in the map return the librairy DEFAULT_PIPELINE.
+    fn pipeline_from<'a>(&'a self, text: &'a str) -> &Pipeline {
```
Comment (MarinPostma) [inline src/tokenizer.rs:154]: same here

**Review by MarinPostma — CHANGES_REQUESTED** [review]

**Review by MarinPostma — APPROVED** [review]

**Pull request merged**